MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/XmpParser.java

```diff
@@ -0,0 +1,46 @@
+package com.google.android.libraries.motionphotoreader;
+
+import androidx.annotation.Nullable;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+import com.adobe.internal.xmp.XMPMetaFactory;
+import com.google.common.io.ByteStreams;
+import com.google.common.primitives.Bytes;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * The XmpParser class is a package-private class intended to help extract the microvideo offset
+ * information from the XMP metadata of the given Motion Photo. This class is for internal use only.
+ */
+class XmpParser {
+
+    private static final byte[] OPEN_ARR = "<x:xmpmeta".getBytes();  /* Start of XMP metadata tag */
+    private static final byte[] CLOSE_ARR = "</x:xmpmeta>".getBytes();  /* End of XMP metadata tag */
+
+    /**
+     * Returns the metadata of the Motion Photo file.
```

Comment:
update comment:
```
Returns the metadata of the motion photo file or null if the metadata is missing
```

Reply:
I refactored the XmpParser class slightly to help write some unit tests, so the return statement doesn't return null if the metadata is missing anymore but throws an XMPException if something goes wrong. Otherwise, an XMPMeta object will be returned.